src/SSHDebugPS/ConnectionManager.cs

```diff
@@ -82,19 +84,48 @@ public static SSHConnection GetSSHConnection(string name)
                 IVsConnectionManager connectionManager = (IVsConnectionManager)ServiceProvider.GlobalProvider.GetService(typeof(IVsConnectionManager));
                 if (connectionManager != null)
                 {
-                    IConnectionManagerResult result;
+                    IConnectionManagerResult result = null;
                     if (string.IsNullOrWhiteSpace(name))
                     {
                         result = connectionManager.ShowDialog();
                     }
                     else
                     {
-                        ParseSSHConnectionString(name, out string userName, out string hostName, out int port);
+                        ParseSSHConnectionString(name, out string userName, out SecureString password, out string hostName, out int port);
 
-                        result = connectionManager.ShowDialog(new PasswordConnectionInfo(hostName, port, Timeout.InfiniteTimeSpan, userName, new System.Security.SecureString()));
+                        if (password == null)
+                        {
+                            result = connectionManager.ShowDialog(new PasswordConnectionInfo(hostName, port, Timeout.InfiniteTimeSpan, userName, new SecureString()));
+                        }
+                        else
+                        {
+                            connectionInfo = new PasswordConnectionInfo(hostName, port, Timeout.InfiniteTimeSpan, userName, password);
+
+                            // Attempt to open the temp connection so that the host key can be verified.
+                            try
+                            {
+                                using (var system = new UnixSystem())
+                                {
+                                    system.Connect(connectionInfo);
+                                }
+                            }
+                            catch (RemoteConnectivityException e) when (!String.IsNullOrEmpty(e.Fingerprint))
+                            {
```

Comment:
>{ [](http://example.com/codeflow?start=28&length=1)

I think we need to throw if we aren't on the UI thread.

Reply:
First line in this method does that.

Reply:
Thanks. Somehow missed that.